parfive/downloader.py

```diff
@@ -661,7 +675,11 @@ async def _http_download_worker(self, session, url, chunksize, http_range, timeo
             while True:
                 chunk = await resp.content.read(chunksize)
                 if not chunk:
-                    break
+                    if resp.content_length is not None:
+                        if resp.content_length - offset != 0:
+                            raise Exception("Content length mismatch")
```

Comment:
This is probably wrong and useless.

No idea how to mock a chunk failing either.

Reply:
What's this trying to achieve? I think if any error happens in reading the bytes then the `read()` call will error, I don't *think* we need to explicitly check this here.